### PR TITLE
Fire centre drop down

### PR DESCRIPTION
**Pull request by conbrad**
- Adds `api/fba/fire-centers` route
- Adds `fire_center_mapper` to marshall station results from WFWX1 into unque `FireCenter`s with their associated stations to serve to the frontend
- Selected a fire center on the frontend renders the list of associated stations

Naming note: Distinguishing this new page with the name `fba` as opposed to the existing `fba-calc`. Not sure if there is a better name yet.

# Test Links:
[Percentile Calculator](https://wps-pr-1483.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1483.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1483.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1483.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1483.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1483.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1483.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Review by tessmgood — APPROVED** [review]
there is indeed a dropdown! exciting 
fba is a good name for now - it exactly matches the name of the document that this tool is ultimately producing. we'll see how users & stakeholders interpret it :)

File: api/app/routers/fba.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+router = APIRouter(
+    prefix="/fba",
+    dependencies=[Depends(authentication_required), Depends(audit)],
+)
+
+
+@router.get('/fire-centers/', response_model=FireCenterListResponse)
+async def get_all_fire_centers(response: Response,
+                         _=Depends(authentication_required)):
+    """ Returns daily metrics for each station code. """
```
Comment (Sybrand) [inline api/app/routers/fba.py:21]: This comment isn't relevant

File: api/app/routers/fba.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    try:
+        logger.info('/fba/fire-centers/')
+        response.headers["Cache-Control"] = "max-age=0"  # don't let the browser cache this
```
Comment (Sybrand) [inline api/app/routers/fba.py:24]: Why don't we cache this? It shouldn't change often.

File: api/app/tests/test_fba_endpoint.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+from starlette.testclient import TestClient
+from aiohttp import ClientSession
+from app.schemas.fba import FireCenterStation, FireCentre
+from app.tests.common import default_mock_client_get
+import app.main
+from app.tests import load_json_file
+import app.routers.hfi_calc
+from app.tests.test_hfi_dailies import AsyncIter
+
+fire_centers = AsyncIter([
```
Comment (Sybrand) [inline api/app/tests/test_fba_endpoint.py:13]: Is this being used?

File: api/app/wildfire_one/wfwx_api.py [hunk 4]
```diff
@@ -299,3 +300,9 @@ async def get_dailies(
         cache_expiry_seconds=cache_expiry_seconds)
 
     return dailies_iterator
+
+
+async def get_fire_centers(session: ClientSession, header: dict,) -> List[FireCentre]:
+    """ Get the fire centers. Replace hard coded centers once WFWX API supports them """
```
Comment (Sybrand) [inline api/app/wildfire_one/wfwx_api.py:306]: Doesn't look hard coded to me?

**Review by Sybrand — CHANGES_REQUESTED** [review]

File: web/src/features/fbaCalculator/components/FormalFBATable.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  testId?: string
+  className: string
+  fireCenter: FireCenter
+}
+
+const buildAdvisoryHeader = (fireCenter: FireCenter) => (
+  <h2>{fireCenter.name}: Fire Behaviour Advisory Summary</h2>
+)
+
+const FormalFBATable = (props: FormalFBATableProps) => {
```
Comment (andrea-williams) [inline web/src/features/fbaCalculator/components/FormalFBATable.tsx:16]: I'm not a big fan of the name of this component, but until we figure out specific names for the 2 FBA-related apps, I'll go with this

**Review by andrea-williams — APPROVED** [review]

File: api/app/wildfire_one/schema_parsers.py [hunk 6]
```diff
@@ -68,6 +69,28 @@ async def wfwx_station_list_mapper(raw_stations: Generator[dict, None, None]) ->
...
+                                        name=raw_station['displayLabel'],
+                                        zone=construct_zone_code(raw_station))
+
+            fire_center = fire_centers.get(fire_center_id, None)
+            if fire_center is None:
+                fire_centers[fire_center_id] = FireCentre(
+                    id=raw_fire_center['id'], name=raw_fire_center['displayLabel'], stations=[station])
+            else:
+                fire_center.stations.append(station)
+    return fire_centers
```
[the remark this answers is not in the document]
Comment (conbrad) [reply api/app/wildfire_one/schema_parsers.py:91]: Let's discuss and do a find/replace follow up PR.

File: api/app/routers/fba.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (conbrad) [reply api/app/routers/fba.py]: Copy/paste, already changed since Sy caught it earlier.

**Review by JulianForeman — APPROVED** [review]

**Pull request merged**